tests/scattering1d/test_jax_backend_1d.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+from kymatio import Scattering1D
+import os
+import numpy
+import jax.numpy as np
+from jax import device_put
+import io
+
+backends = []
+
+from kymatio.scattering1d.backend.jax_backend import backend
+backends.append(backend)
+
+@pytest.mark.parametrize('backend', backends)
+def test_Scattering1D(backend):
+    """
+    Applies scattering on a stored signal to make sure its output agrees with
+    a previously calculated version.
+    """
+    test_data_dir = os.path.dirname(__file__)
+
+    with open(os.path.join(test_data_dir, 'test_data_1d.npz'), 'rb') as f:
+        buffer = io.BytesIO(f.read())
+        data = numpy.load(buffer)
+
+    x = device_put(np.asarray(data['x']))
+    J = data['J']
+    Q = data['Q']
+    Sx0 = device_put(np.asarray(data['Sx']))
+
+    T = x.shape[-1]
+    scattering = Scattering1D(J, T, Q, backend=backend, frontend='jax')
+
+    Sx = scattering(x) 
```

Comment:
Do we know that Jax is actually being used here (as opposed to NP)? There should be a way to find out using the mock package, I think. Conversely, it would be good to make sure that NP *is not* being used (except for filter generation) so that a call doesn't slip through by accident.

Reply:
![image](https://user-images.githubusercontent.com/1306635/124052580-b9581980-d9ec-11eb-9bd2-e95dbfff15f6.png)

Here is another pictorial proof.

You mean build a fake numpy or something that errors on all/most methods? With https://docs.python.org/dev/library/unittest.mock.html ?

Reply:
> You mean build a fake numpy or something that errors on all/most methods?

Yeah or is that overkill, do you think? You can just do things like `.called` on various methods. Maybe NP will crash already if you give it Jax arrays.